pkg/syncer/tunneler.go

```diff
@@ -126,3 +152,132 @@ func startTunneler(ctx context.Context, upstream, downstream *rest.Config, syncT
 	logger.V(2).Info("stop serving on reverse connection")
 	return err
 }
+
+func proxyHandlerFiltered(proxy *httputil.ReverseProxy, listerFunc ListerFuncForResource, synctargetClusterName logicalcluster.Name, synctargetName, syncTargetUID string) http.HandlerFunc {
```

Comment:
Can you take a `http.handler` here as first argument, instead of a `ReverseProxy` ? It seems we're simply using a Handler here. No need to depend on the fact it is a proxy.

The function itself could then be something like `withPodAccessCheck(handler)`

Reply:
done